### ISSUE TITLE
[java] fix-pom.py should not create versions.txt in a module in the Java monorepo

### ISSUE THREAD

**Issue by suztomo** [googleapis/synthtool#1707]
The Java monorepo is now starting to use the versions.txt at the root for all submodules in the repository (single-component setup).

In https://github.com/googleapis/google-cloud-java/pull/8731/files, owlbot-java unexpectedly created "versions.txt" in the eventarc module and reset the version to 0.0.1-SNAPSHOT.

![image](https://user-images.githubusercontent.com/28604/201212508-5fe6c980-ac66-4241-a9be-47a131429d4b.png)

I need to find a way to adjust the single-component setup:

- When an existing module of the google-cloud-java monorepo is updated
- When a new module of google-cloud-java monorepo is created
- When an existing module of a split repo is updated
- When a new module of an existing split repo is created
- When a new split repo library is created (Maybe I don't need this)

# How to confirm?

In one window, run:

```
~/synthtool$  docker build -f docker/owlbot/java/Dockerfile .
```

In another window, run:

```
cd google-cloud-java
gh pr checkout 8731
git reset --hard d9356a599b5a06689bf70c48f482b8ac28744a34
```

And then

```
git reset --hard d9356a599b5a06689bf70c48f482b8ac28744a34
docker run --rm -v $(pwd):/workspace XXXXXXX
chown -R suztomo .
git add -A
git commit -m 'chore: manual owlbot-java invocation'
```

Check java-eventarc/google-cloud-eventarc-bom/pom.xml

**Comment by suztomo** [comment]
The problem is that versions.txt is used to multiple purposes

- Release Please uses it to track versions
- owlbot-java uses it to decide the member of the BOM

**Comment by suztomo** [comment]
It's resolved by #1706
